src/processor/logic.py

```diff
@@ -194,3 +194,28 @@ def get_language_from_wiki_url(url, default_language='en'):
         return match.group()[2:4]
 
     return default_language
+
+
+def set_generic_twitter_link(tweet_id):
+    """ Convert tweet ID or URL to a generic URL to view the tweet. If tweet_id
+    is given as a URL, the function will try to determine what form of URL is
+    given and extract the tweet ID accordingly.
+
+    Args:
+        tweet_id (str): ID/URL of a tweet
+
+    Returns:
+        str: url to view the tweet
+    """
+    if not tweet_id.isdigit():
+        split_url = urlsplit(tweet_id)
+        if split_url.query:
+            try:  # extract tweet ID from URL paramters
+                tweet_id = parse_qs(split_url.query)['id'][0]
+            except (KeyError, IndexError):
```

Comment:
I think you need to log the exception as well, as logging the fact there was an error with such and such tweet id, is not helpful to those debugging what's going wrong, if anything does go wrong.

Reply:
Well spotted! Thanks. I actually want this to raise an error for now. 
Will update accordingly.